File: TP1/troca.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <string.h>

void substituirLetra(char *texto, char letra1, char letra2) {
    for (int i = 0; texto[i] != '\0'; i++) {
        if (texto[i] == letra1) {
            texto[i] = letra2;
        }
    }
}

int main() {
    srand(time(NULL));

    while (1) {
       char entrada[100];
        fgets(entrada, sizeof(entrada), stdin);

        if (strncmp(entrada, "FIM", 3) == 0) {
            break;
        }

        char letra1 = 'a' + (rand() % 26);
        char letra2 = 'a' + (rand() % 26);

        substituirLetra(entrada, letra1, letra2);
        printf("%s", entrada);
    }

    return 0;
}

File: TP1/troca.java
import java.util.Random;
import java.util.Scanner;

public class troca {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Random gerador = new Random();
        gerador.setSeed(4);
        String entrada = scanner.nextLine();

        while (!entrada.equals("FIM")) {
            //System.out.print("Digite uma frase (ou 'FIM' para encerrar): ");
            


            char letra1 = (char) ('a' + (Math.abs(gerador.nextInt()) % 26));
            char letra2 = (char) ('a' + (Math.abs(gerador.nextInt()) % 26));
            String resultado = substituirLetra(entrada, letra1, letra2);
            System.out.println(resultado);
            entrada = scanner.nextLine();
        }

        scanner.close();
    }

    public static String substituirLetra(String texto, char letra1, char letra2) {
        char[] caracteres = texto.toCharArray();
        for (int i = 0; i < caracteres.length; i++) {
            if (caracteres[i] == letra1) {
                caracteres[i] = letra2;
            }
        }
        return new String(caracteres);
    }
}
